### ISSUE TITLE
Handle Ledger devices

### ISSUE THREAD

**Issue by tombeynon** [eco-stake/restake#144]
I'm learning that when you send a TX you can provide it in two structures; Amino (legacy) and Protobuf. Keplr uses Protobuf if the wallet supports it, otherwise falls back to Amino. Typically transactions are sent in Protobuf format, and the blockchain (or Keplr, CosmJS..?) automatically converts them to Amino if it knows how to do so.

Authz transactions aren't supported for this conversion currently and REStake only send the Protobuf version. More importantly though, it doesn't seem like it's possible to send the right message structure based on the discussion in the below issues.

I need to spend a bit of time working out if it's possible to work around this, or if we have to wait for a chain upgrade to an unreleased SDK version.

Related:

https://github.com/cosmos/cosmjs/issues/1026

https://github.com/cosmos/cosmos-sdk/issues/11190

https://github.com/cosmos/cosmos-sdk/pull/11224

**Comment by tombeynon** [comment]
Specific comment about working around with cosmjs https://github.com/cosmos/cosmjs/issues/1026#issuecomment-1043541379

**Comment by webmaster128** [comment]
Please see https://medium.com/confio/authz-and-ledger-signing-an-executive-summary-8754a0dc2a88 for a high level overview of the current limitation.

(Props for the work here by the way!)

**Comment by tombeynon** [comment]
Thanks so much for the writeup @webmaster128, that's extremely helpful. Is there a rough ETA on SDK 0.46? Would much rather wait for that than try to hack around it, but as you can imagine, 'wen ledger' is getting louder 😂

**Comment by webmaster128** [comment]
I hear rumors of 1-2 months, but I'm not involved in Cosmos SDK development. Certainly this summer.

**Comment by tombeynon** [comment]
Okay great. Thanks for everything you do @webmaster128, relied on a lot of your work to build this

**Comment by webmaster128** [comment]
Which message types exactly do you need? This issue was pulled out and is still open: https://github.com/cosmos/cosmjs/issues/1092

Which networks migrated to 0.46 already?

**Comment by tombeynon** [comment]
Resolved in #597. Secret Network mainnet is the first with enabled support, and new chains will be enabled as required.

A separate task needs to be completed to identify and enable v0.46 chains automatically but we have some time for this.

The Authz Amino converter is the solution if anyone finds this looking for implementation details - https://github.com/eco-stake/restake/blob/master/src/converters/Authz.mjs

**Comment by assafmo** [comment]
To identify the Cosmos SDK version you can use the `/cosmos/base/tendermint/v1beta1/node_info` endpoint, e.g.:
```bash
curl https://secret-4.api.trivium.network:1317/cosmos/base/tendermint/v1beta1/node_info -s | jq '.application_version.cosmos_sdk_version'
```
